test/TestUserAcceptance.hs

```diff
@@ -282,16 +279,35 @@ assertRegex regex out = liftIO $ assertBool
 defaultSessionName :: String
 defaultSessionName = "purebredtest"
 
-tearDown :: (String, String, String) -> IO ()
-tearDown (testdir, _, _)= do
+data Env = Env
+  { _envDir :: FilePath
+  , _envMaildir :: FilePath
+  , _envSessionName :: String
+  }
+
+envDir :: Lens' Env FilePath
+envDir f (Env a b c) = fmap (\a' -> Env a' b c) (f a)
+
+envMaildir :: Lens' Env FilePath
+envMaildir f (Env a b c) = fmap (\b' -> Env a b' c) (f b)
+
+envSessionName :: Lens' Env String
+envSessionName f (Env a b c) = fmap (\c' -> Env a b c') (f c)
+{-# ANN envSessionName ("HLint: ignore Avoid lambda" :: String) #-}
+
+-- | Tear down a test session
+tearDown :: Env -> IO ()
+tearDown (Env testdir _ sessionName) = do
   removeDirectoryRecursive testdir
-  cleanUpTmuxSession defaultSessionName
+  cleanUpTmuxSession sessionName
 
-setUp :: IO (String, String, String)
-setUp = do
-  let sessionname = defaultSessionName
-  (testdir, testmdir) <- setUpTmuxSession sessionname >> setUpTempMaildir
-  pure (testdir, testmdir, sessionname)
+-- | Set up a test session.
+setUp :: Int -> IO Env
+setUp i = do
+  let sessionName = defaultSessionName <> show i
```

Comment:
That would create sessions like `purebredtest1,2,3,4...` right. You do have the test case name, why not use that as an additional identifier? That'll help when debugging any failures.

Reply:
I'll do this in a separate commit.

Reply:
Fair enough

Reply:
PR: https://github.com/purebred-mua/purebred/pull/145